go/cmd/dolt/commands/diff_output.go

```diff
@@ -390,9 +450,138 @@ func (j *jsonDiffWriter) RowWriter(ctx context.Context, td diff.TableDelta, unio
 	return j.rowDiffWriter, err
 }
 
+func (j *jsonDiffWriter) WriteTriggerDiff(ctx context.Context, triggerName, oldDefn, newDefn string) error {
+	if j.triggersWritten == 0 {
+		// begin the document if necessary
+		if j.tablesWritten == 0 {
+			_, err := j.wr.Write([]byte("{"))
+			if err != nil {
+				return err
+			}
+		} else {
+			_, err := j.wr.Write([]byte(jsonDataDiffFooter + ","))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if j.triggersWritten == 0 {
+		_, err := j.wr.Write([]byte(`"triggers":[`))
+		if err != nil {
+			return err
+		}
+	} else {
+		_, err := j.wr.Write([]byte(","))
+		if err != nil {
+			return err
+		}
+	}
+
+	triggerNameBytes, err := ejson.Marshal(triggerName)
+	if err != nil {
+		return err
+	}
+
+	oldDefnBytes, err := ejson.Marshal(oldDefn)
+	if err != nil {
+		return err
+	}
+
+	newDefnBytes, err := ejson.Marshal(newDefn)
+	if err != nil {
+		return err
+	}
+
+	_, err = j.wr.Write([]byte(fmt.Sprintf(`{"name":%s,"from_definition":%s,"to_definition":%s}`,
+		triggerNameBytes, oldDefnBytes, newDefnBytes)))
+	if err != nil {
+		return err
+	}
+
+	j.triggersWritten++
+	return nil
+}
+
+func (j *jsonDiffWriter) WriteViewDiff(ctx context.Context, viewName, oldDefn, newDefn string) error {
+	if j.viewsWritten == 0 {
+		// begin the document if necessary
+		if j.tablesWritten == 0 && j.triggersWritten == 0 {
+			_, err := j.wr.Write([]byte("{"))
+			if err != nil {
+				return err
+			}
+		}
+
+		// end the previous block if necessary
+		if j.tablesWritten > 0 && j.triggersWritten == 0 {
+			_, err := j.wr.Write([]byte(jsonDataDiffFooter + ","))
```

Comment:
Again, this seems frail. Maybe we shouldn't be building this up as a stream, but as objects in a hierarchy. Then you can wrap junks in brackets and join them all together with ',' and the separator.

Reply:
It's a little better with the beginDocument method, but still pretty frail. Not going to restructure it that much right now though.